packages/smooth_app/lib/pages/onboarding/onboarding_flow_navigator.dart

```diff
@@ -24,193 +24,143 @@ enum OnboardingPage {
       case OnboardingPage.CONSENT_PAGE:
-        return false;
+        return const Color(0xFFFFF2DF);
       case OnboardingPage.ONBOARDING_COMPLETE:
-        return true;
-    }
-  }
-
-  void navigateToPage(BuildContext context, OnboardingPage page) {
-    _userPreferences.setLastVisitedOnboardingPage(page);
-    _historyOnboardingNav.add(page);
-
-    final MaterialPageRoute<Widget> route = MaterialPageRoute<Widget>(
-      builder: (BuildContext context) => getPageWidget(context, page),
-    );
-
-    if (page == OnboardingPage.ONBOARDING_COMPLETE) {
-      Navigator.pushAndRemoveUntil(
-        context,
-        route,
-        (Route<dynamic> route) => false,
-      );
-    } else {
-      Navigator.push<Widget>(context, route);
+        // whatever, it's not used
+        return Colors.black;
     }
   }
 
-  Widget getPageWidget(BuildContext context, OnboardingPage page) {
+  Widget getPageWidget(BuildContext context) {
     final LocalDatabase localDatabase = context.read<LocalDatabase>();
-    switch (page) {
+    final Color backgroundColor = getBackgroundColor();
+    switch (this) {
       case OnboardingPage.NOT_STARTED:
       case OnboardingPage.REINVENTION:
-        return ReinventionPage(getBackgroundColor(page));
+        return ReinventionPage(backgroundColor);
       case OnboardingPage.WELCOME:
-        return WelcomePage(getBackgroundColor(page));
+        return WelcomePage(backgroundColor);
       case OnboardingPage.SCAN_EXAMPLE:
         return _wrapWidgetInCustomBackNavigator(
           context,
-          page,
-          ScanExample(getBackgroundColor(page)),
+          ScanExample(backgroundColor),
         );
       case OnboardingPage.HEALTH_CARD_EXAMPLE:
         return _wrapWidgetInCustomBackNavigator(
           context,
-          page,
-          SampleHealthCardPage(localDatabase, getBackgroundColor(page)),
+          SampleHealthCardPage(localDatabase, backgroundColor),
         );
       case OnboardingPage.ECO_CARD_EXAMPLE:
         return _wrapWidgetInCustomBackNavigator(
           context,
-          page,
-          SampleEcoCardPage(localDatabase, getBackgroundColor(page)),
+          SampleEcoCardPage(localDatabase, backgroundColor),
         );
       case OnboardingPage.PREFERENCES_PAGE:
         return _wrapWidgetInCustomBackNavigator(
           context,
-          page,
-          PreferencesPage(localDatabase, getBackgroundColor(page)),
+          PreferencesPage(localDatabase, backgroundColor),
         );
       case OnboardingPage.PERMISSIONS_PAGE:
         return _wrapWidgetInCustomBackNavigator(
           context,
-          page,
-          PermissionsPage(getBackgroundColor(page)),
+          PermissionsPage(backgroundColor),
         );
       case OnboardingPage.CONSENT_PAGE:
         return _wrapWidgetInCustomBackNavigator(
           context,
-          page,
-          ConsentAnalyticsPage(getBackgroundColor(page)),
+          ConsentAnalyticsPage(backgroundColor),
```

Comment:
I know it's not from you but I am wondering why don't we just have the color inside the page